### PR TITLE
Add pipeline pause and build duration

### PR DESCRIPTION
**Pull request by FlorianVeaux**
Superseding: https://github.com/jenkinsci/datadog-plugin/pull/57 but keeping the commit history.
Fixed conflicts, improved dependencies management using the jenkins BOM

Original description:

> ## What does this PR do?
> Currently the build duration is the total time a job takes to build - including "waiting for approval". If we would use that metric to show off we would see some builds where a long build duration is shown but no one will understand that this is not only the time to build the code but also waiting steps. Therefore we want to have the possibility to also show the duration how long each job was paused.
This PR will introduce the pause duration and duration without pause duration as new metric.
I also added some log messages so that the durations are also visible in the jenkins log if the specified logger is FINE enabled.

>## Verification Process
> I ran all tests and they are all fine
I checked the introduced log messages in my jenkins in order to verify that the timings are correct
I installed the plugin on my jenkins, ran a job and verified on datadog, that all data is there as expected

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java [hunk 1]
```diff
@@ -241,6 +267,13 @@ public void onDeleted(Run run) {
         }
     }
 
+    private String toTimeString(long millis) {
+        return String.format("%d min, %d sec",
+                TimeUnit.MILLISECONDS.toMinutes(millis),
+                TimeUnit.MILLISECONDS.toSeconds(millis) -
+                        TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(millis)));
+    }
```
Comment (florimondmanca) [inline src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:275]: (Nit) Using intermediary variables for readability (not sure if `long` is the correct data type):

```suggestion
        long minutes = TimeUnit.MILLISECONDS.toMinutes(millis);
        long totalSeconds = TimeUnit.MILLISECONDS.toSeconds(millis);
        long seconds = totalSeconds - TimeUnit.MINUTES.toSeconds(minutes);
        return String.format("%d min, %d sec", minutes, seconds);
    }
```

File: src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java [hunk 2]
```diff
@@ -154,6 +160,21 @@ public void onCompleted(Run run, @Nonnull TaskListener listener) {
...
             client.gauge("jenkins.job.duration", buildData.getDuration(0L) / 1000, hostname, tags);
+            logger.fine(String.format("[%s]: Duration: %s", buildData.getJobName(null), toTimeString(buildData.getDuration(0L))));
+
+            if (run instanceof WorkflowRun) {
+                RunExt extRun = getRunExtForRun((WorkflowRun) run);
+                long pauseDuration = 0;
+                for (StageNodeExt stage : extRun.getStages()) {
+                    pauseDuration += stage.getPauseDurationMillis();
+                }
+                client.gauge("jenkins.job.pauseduration", pauseDuration / 1000, hostname, tags);
```
Comment (sarah-witt) [inline src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:171]: to match the stage duration names we should update all instances of these metrics like so:
```suggestion
                client.gauge("jenkins.job.pause_duration", pauseDuration / 1000, hostname, tags);
```

Comment (FlorianVeaux) [reply src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:171]: Sure, makes sense 👍

File: README.md [hunk 3]
```diff
@@ -214,13 +214,15 @@ NOTE: `event_type` is always set to `security` for above events and metrics.
 | `jenkins.item.location_changed`        | Rate of items being moved.                                     | `jenkins_url`, `user_id`                                    |
 | `jenkins.item.updated`                 | Rate of items being updated.                                   | `jenkins_url`, `user_id`                                    |
 | `jenkins.job.aborted`                  | Rate of aborted jobs.                                          | `branch`, `jenkins_url`, `job`, `node`, `user_id`           |
+| `jenkins.job.buildduration`            | Build duration without pause (in seconds).                     | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
```
Comment (sarah-witt) [inline README.md:217]:
```suggestion
| `jenkins.job.build_duration`            | Build duration without pause (in seconds).                     | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
```

File: README.md [hunk 4]
```diff
@@ -214,13 +214,15 @@ NOTE: `event_type` is always set to `security` for above events and metrics.
...
 | `jenkins.job.cycletime`                | Build Cycle Time.                                              | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
 | `jenkins.job.duration`                 | Build duration (in seconds).                                   | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
 | `jenkins.job.feedbacktime`             | Feedback time from code commit to job failure.                 | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
 | `jenkins.job.leadtime`                 | Build Lead Time.                                               | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
 | `jenkins.job.mtbf`                     | MTBF, time between last successful job and current failed job. | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
 | `jenkins.job.mttr`                     | MTTR: time between last failed job and current successful job. | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
+| `jenkins.job.pauseduration`            | Pause duration of build job (in seconds).                      | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
```
Comment (sarah-witt) [inline README.md:225]:
```suggestion
| `jenkins.job.pause_duration`            | Pause duration of build job (in seconds).                      | `branch`, `jenkins_url`, `job`, `node`, `result`, `user_id` |
```

**Pull request merged**